### PR TITLE
Deprecate camelCase parameters used by SAML APIs

### LINKED ISSUE #73543: SAML complete logout API has a camel case request parameter
As discussed in https://github.com/elastic/elasticsearch/pull/72867/files#r629011079, the SAML complete logout API has a camel case request parameter, `queryString`. This is an anomaly because the rest of APIs use only snake case for request parameters. 

This request parameter, `queryString`, should be deprecated and replaced with its snake case equivalent, `query_string`.

### PR DESCRIPTION
**Pull request by ywangd**
Both [SAML complete logout](https://www.elastic.co/guide/en/elasticsearch/reference/7.x/security-api-saml-complete-logout.html) and [SAML invalidate session](https://www.elastic.co/guide/en/elasticsearch/reference/7.x/security-api-saml-invalidate.html) APIs use a camelCase request parameter, `queryString`, while the convention is to use snake_case parameters. This PR deprecates `queryString` and replaces it with `query_string`. It is an error to if a request specifies both of them.

Resolves: #73543

### CODE REVIEW

**Comment by tvernum** [comment]
Ping @elastic/kibana-security.

We're considering deprecating the current parameter you use, and switching it to `snake_case`. 
I think we should coordinate changes, so that you can update the Kibana code in the same release and customers don't see deprecation warnings.

**Comment by azasypkin** [comment]
Thanks for the ping. Sure, we'll track this PR and update Kibana code base for 7.14 as soon as updated ES snapshot reaches our CI.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/saml/RestSamlCompleteLogoutAction.java [hunk 1]
```diff
@@ -47,10 +47,12 @@
         PARSER = new ObjectParser<>("saml_complete_logout", SamlCompleteLogoutRequest::new);
 
     static {
-        PARSER.declareStringOrNull(SamlCompleteLogoutRequest::setQueryString, new ParseField("queryString"));
+        PARSER.declareStringOrNull(SamlCompleteLogoutRequest::setQueryString,
+            new ParseField("queryString").withAllDeprecated("query_string"));
         PARSER.declareStringOrNull(SamlCompleteLogoutRequest::setContent, new ParseField("content"));
         PARSER.declareStringArray(SamlCompleteLogoutRequest::setValidRequestIds, new ParseField("ids"));
         PARSER.declareString(SamlCompleteLogoutRequest::setRealm, new ParseField("realm"));
+        PARSER.declareStringOrNull(SamlCompleteLogoutRequest::setQueryString, new ParseField("query_string"));
```
Comment (tvernum) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/saml/RestSamlCompleteLogoutAction.java:55]: This seems weird. Can't you just use the deprecated names option on a single field?
```suggestion
        PARSER.declareStringOrNull(SamlCompleteLogoutRequest::setQueryString, new ParseField("query_string", "queryString"));
        PARSER.declareStringOrNull(SamlCompleteLogoutRequest::setContent, new ParseField("content"));
        PARSER.declareStringArray(SamlCompleteLogoutRequest::setValidRequestIds, new ParseField("ids"));
        PARSER.declareString(SamlCompleteLogoutRequest::setRealm, new ParseField("realm"));
```

Comment (ywangd) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/saml/RestSamlCompleteLogoutAction.java:55]: 👍

**Comment by ywangd** [comment]
@elasticmachine update branch

**Review by tvernum — APPROVED** [review]
LGTM

**Pull request merged**